test/system/test_tpm2_create.sh

```diff
@@ -53,7 +53,7 @@ tpm2_createprimary -Q -A p -g sha1 -G rsa -C context.out
 
 # Keep the algorithm specifiers mixed to test friendly and raw
 # values.
-for gAlg in sha1 0x0B sha384; do
+for gAlg in sha1 0x0B sha256; do
```

Comment:
The test is already testing for sha256, is just using the hex value (0x0B) since both pretty names and hex values are supported. So your change is testing sha256 twice. So just remove sha384 and keep sha1 and 0x0B (we are using both pretty names and hex values on purpose to make sure that the parse logic isn't regressed).

Your 0x2c3 error code is indeed TPM_RC_HASH (hash algorithm not supported or not appropriate). So if this hash algorithm is not common on real TPMs, then I agree with you that the test could omit it.

In your commit message please mention issue #335 (Make sure that all tests work on a real TPM hardware) since this change also help on that regard.

Reply:
A long term solution for this should be to query the TPM about the supported algorithms and only test those, i.e:

for gAlg in $(tpm2_dump_capability -c algorithms | grep sha | awk '/TPMA_ALGORITHM for ALG_ID:/{print $4}'); do